python/dgl/contrib/unified_tensor.py

```diff
@@ -0,0 +1,75 @@
+"""Unified Tensor."""
+from .. import backend as F
+from .._ffi.function import _init_api
+from .. import utils
+
+
+class UnifiedTensor: #UnifiedTensor
+    '''Class for storing unified tensor.
+
+    Parameters
+    ----------
+    input : Tensor
+        Tensor which we want to convert into the
+        unified tensor.
+    device : device
+        Device to create the mapping of the unified tensor.
+    '''
+
+    def __init__(self, input, device):
+        if F.device_type(device) != 'cuda':
+            raise ValueError("Target device must be a cuda device")
+
+        self._input = input
+        self._array = F.zerocopy_to_dgl_ndarray(self._input)
```

Comment:
Do we need to make sure the `input` is pinned in CPU memory? If so we need to highlight it in the doc string.

Reply:
No problem. Added more explanation in the doc string. __init__ process includes pinning the memory so the users do not need to worry about it.